src/services/routes/org.routes.spec.ts

```diff
@@ -186,7 +188,147 @@ describe('org routes', () => {
+      method: 'PATCH',
+      url: `/orgs/${ORG_NAME}`,
+    };
+    const getOrgSuccessResponse = {
+      didSucceed: true,
+
+      result: {
+        name: ORG_NAME,
+        memberAccessType: 'INVITE_ONLY',
+      },
+    } as const;
+
+    test('Empty parameters should be accepted', async () => {
+      const serverInstance = await makeServer();
+      const payload: OrgSchemaPatch = {};
+      mockGetOrg.mockResolvedValueOnce(getOrgSuccessResponse);
+      mockUpdateOrg.mockResolvedValueOnce({
+        didSucceed: true,
+      });
+
+      const response = await serverInstance.inject({
+        ...injectionOptions,
+        payload,
+      });
+
+      expect(response).toHaveProperty('statusCode', 200);
+    });
+
+    test('name matching the url parameter should be accepted', async () => {
+      const serverInstance = await makeServer();
+      const payload: OrgSchemaPatch = {
+        name: ORG_NAME,
+      };
+      mockGetOrg.mockResolvedValueOnce(getOrgSuccessResponse);
+      mockUpdateOrg.mockResolvedValueOnce({
+        didSucceed: true,
+      });
+
+      const response = await serverInstance.inject({
+        ...injectionOptions,
+        payload,
+      });
+
+      expect(response).toHaveProperty('statusCode', 200);
+    });
+
+    test('INVITE_ONLY access type should be accepted', async () => {
+      const serverInstance = await makeServer();
+      const payload: OrgSchemaPatch = {
+        memberAccessType: 'INVITE_ONLY',
+      };
+      mockGetOrg.mockResolvedValueOnce(getOrgSuccessResponse);
+      mockUpdateOrg.mockResolvedValueOnce({
+        didSucceed: true,
+      });
+
+      const response = await serverInstance.inject({
+        ...injectionOptions,
+        payload,
+      });
+
+      expect(response).toHaveProperty('statusCode', 200);
+    });
+
+    test('OPEN access type should be accepted', async () => {
+      const serverInstance = await makeServer();
+      const payload: OrgSchemaPatch = {
+        memberAccessType: 'OPEN',
+      };
+      mockGetOrg.mockResolvedValueOnce(getOrgSuccessResponse);
+      mockUpdateOrg.mockResolvedValueOnce({
+        didSucceed: true,
+      });
+
+      const response = await serverInstance.inject({
+        ...injectionOptions,
+        payload,
+      });
+
+      expect(response).toHaveProperty('statusCode', 200);
+    });
+
+    test('awala endpoint type should be accepted', async () => {
+      const serverInstance = await makeServer();
+      const payload: OrgSchemaPatch = {
+        awalaEndpoint: AWALA_ENDPOINT,
+      };
+      mockGetOrg.mockResolvedValueOnce(getOrgSuccessResponse);
+      mockUpdateOrg.mockResolvedValueOnce({
+        didSucceed: true,
+      });
+
+      const response = await serverInstance.inject({
+        ...injectionOptions,
+        payload,
+      });
+
+      expect(response).toHaveProperty('statusCode', 200);
+    });
+
+    test('non matching name should resolve into bad request status', async () => {
```

Comment:
I think this test should do two things:

- Check that the response is a 40X with the respective problem type (as it does  right now).
- Check that we haven't called `updateOrg()`.

Consequently, the title would then become something like "Non-matching name should be refused".